jina/orchestrate/pods/__init__.py

```diff
@@ -314,28 +315,31 @@ class Pod(BasePod):
     def __init__(self, args: 'argparse.Namespace'):
         super().__init__(args)
         self.runtime_cls = self._get_runtime_cls()
-        self.worker = multiprocessing.Process(
-            target=run,
-            kwargs={
-                'args': args,
-                'name': self.name,
-                'envs': self._envs,
-                'is_started': self.is_started,
-                'is_shutdown': self.is_shutdown,
-                'is_ready': self.is_ready,
-                'runtime_cls': self.runtime_cls,
-                'jaml_classes': JAML.registered_classes(),
-            },
-            name=self.name,
-            daemon=False,
-        )
+        if __name__ == '__main__':
```

Comment:
add an else with an Exception then

Reply:
not sure it should be an exception.
I think the principle is, if this is not the parent process, don't try to create the child process again from the child process itself. We should do nothing instead. We can surely improve the implementation I just try to confirm it

Reply:
if we do not create `self.worker` everything will crash, so for sure we need to do Exception although I believe this is nothing that will happen, otherwise what u can do is to simply do:

```python
if __name__ == '__main__':
    freeze_support()
#rest of things
```

this way u just do the freeze when it is main and act normally elsewhere